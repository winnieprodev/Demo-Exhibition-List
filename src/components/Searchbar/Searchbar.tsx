import React, { useCallback, useEffect, useRef } from "react";
import { library as iconLibrary } from "@fortawesome/fontawesome-svg-core";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import * as Styled from "./searchbar.styled";

iconLibrary.add(faSearch);

interface OwnProps {
  disabled?: boolean;
  value?: string;
  onChange?: (newValue: string) => void;
  onFocus?: () => void;
  onSearch?: () => void;
  placeholder?: string;
  focus?: boolean;
}

const Searchbar: React.FC<OwnProps> = ({
  disabled = true,
  value,
  onChange,
  onSearch,
  onFocus,
  focus,
  placeholder = "Search Table...",
}) => {
  const onChangeCallback = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      if (onChange) {
        onChange(event.target.value);
      }
    },
    [onChange]
  );

  const onKeyDownCallback = useCallback(
    (event: React.KeyboardEvent) => {
      if (event.key !== "Enter") return;
      if (event.shiftKey) return;
      if (onSearch) {
        event.preventDefault();
    
        onSearch();
      }
    },
    [onSearch]
  );

  const inputRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    if (focus) {
      inputRef?.current?.focus();
    }
  }, [focus, inputRef]);
  
  return (
    <Styled.SearchBar>
      <Styled.Icon>
        <FontAwesomeIcon icon="search" />
      </Styled.Icon>
      <Styled.Input
        aria-label="search-input"
        ref={inputRef}
        disabled={disabled}
        value={value}
        onChange={onChangeCallback}
        onKeyDown={onKeyDownCallback}
        placeholder={placeholder}
        onFocus={onFocus}
      />
    </Styled.SearchBar>
  );
};

export default Searchbar;
