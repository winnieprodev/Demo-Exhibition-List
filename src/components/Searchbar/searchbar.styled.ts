import styled from "styled-components";
import {
  secondaryColorLighter,
} from "styles/colors";

export const SearchBar = styled.div`
  height: 40px;
  display: flex;
  align-items: center;
  background: #fff;
  position: relative;
  transition: background ease 0.2s;
  min-width: 300px;
  max-width: 600px;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 20px;
  margin: auto;
  width: 100%;
  &:hover,
  &:active {
    background: ${secondaryColorLighter};
  }
`;

export const Icon = styled.div`
  font-size: 22px;
  padding-left: 12px;
  margin-right: 12px;
`;

export const Input = styled.input`
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  padding-left: 60px;
  width: calc(100% - 60px);
  outline: none;
  border: none;
  background: none;
`;
