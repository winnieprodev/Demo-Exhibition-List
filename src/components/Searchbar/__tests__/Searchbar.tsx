import { render, fireEvent, cleanup } from "@testing-library/react";

import Searchbar from "../Searchbar";

const setup = () => {
  const utils = render(<Searchbar />)
  const input = utils.getByLabelText('search-input')
  return {
    input,
    ...utils,
  }
}

afterEach(cleanup)

describe("Searchbar", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <Searchbar value="1" onChange={jest.fn()} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Test SearchbarComponent when the value is changed', () => {
      const {input, getByDisplayValue} = setup()
      fireEvent.change(input, {target: {value: '$23.0'}})
      expect(getByDisplayValue("$23.0")).toBeInTheDocument();
    })
  });
});
