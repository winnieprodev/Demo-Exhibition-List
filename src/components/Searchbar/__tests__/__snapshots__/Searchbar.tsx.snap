// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar Snapshot tests it matches the snapshot 1`] = `
.c0 {
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #fff;
  position: relative;
  -webkit-transition: background ease 0.2s;
  transition: background ease 0.2s;
  min-width: 300px;
  max-width: 600px;
  border: 1px solid rgba(0,0,0,.5);
  border-radius: 20px;
  margin: auto;
  width: 100%;
}

.c0:hover,
.c0:active {
  background: #F3F3F3;
}

.c1 {
  font-size: 22px;
  padding-left: 12px;
  margin-right: 12px;
}

.c2 {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  padding-left: 60px;
  width: calc(100% - 60px);
  outline: none;
  border: none;
  background: none;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-search fa-w-16 "
      data-icon="search"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
      />
    </svg>
  </div>
  <input
    aria-label="search-input"
    class="c2"
    disabled=""
    placeholder="Search Table..."
    value="1"
  />
</div>
`;
